include/kodlab_mjbots_sdk/pid_controller.h

```diff
@@ -0,0 +1,533 @@
+/**
+ * @file pid.h
+ * @author Chandravaran Kunjeti (kunjeti@seas.upenn.edu)
+ * @author Raghavesh Viswanath (vrag@seas.upenn.edu)
+ * @brief Class implementation of PID
+ * @date 5/18/2023
+ * @copyright 2023 The Trustees of the University of Pennsylvania.
+ * All rights reserved.
+ *
+ */
+
+#pragma once
+
+#include <math.h>
+#include <array>
+#include <vector>
+#include <algorithm>
+#include <Eigen/Dense>
+#include <tuple>
+
+namespace kodlab {
+
+/**
+ * @brief PIDController class that allows you to use PID control.
+ */
+
+template <typename Scalar, int N=1>
+class PIDController {
+
+public:
+  typedef Eigen::Matrix<Scalar,N,1> VectorNS;
+  
+  /**
+   * @brief Construct a vector Pid object
+   * @param p_gain Proportional Gain
+   * @param d_gain Derivative Gain
+   * @param i_gain Integral Gain
+   * @param time_step Time difference between 2 updates
+   */
+  PIDController(const VectorNS & p_gain,const VectorNS & d_gain,const VectorNS & i_gain, double time_step=0.001,
+    VectorNS deadband = 0.5*VectorNS::Ones(), VectorNS windup_limit = std::numeric_limits<Scalar>::max()*VectorNS::Ones(),
+    std::array<Scalar, 2> saturation = { -std::numeric_limits<Scalar>::max(), std::numeric_limits<Scalar>::max() })
+    : 
+      deadband_(deadband),
+      windup_limit_(windup_limit),
```

Comment:
Let's rename anything with the word windup to accumulator
and then in the comment describing the member variable, add a comment that says
"Used to prevent runaway integrator/windup"